src/herder/TxSetFrame.cpp

```diff
@@ -186,21 +186,23 @@ TxSetFrame::surgePricingFilter(LedgerManager const& lm)
     }
 }
 
-// TODO.3 this and checkValid share a lot of code
-void
-TxSetFrame::trimInvalid(Application& app,
-                        std::vector<TransactionFramePtr>& trimmed)
+template <typename Lambda1, typename Lambda2, typename Lambda3>
```

Comment:
you don't need the template here: lambdas have the same signature or is that to just to try to get better optimized code? Either way, I don't think it's necessary, the compiler will inline if it needs to